variables.tf

```diff
@@ -31,6 +31,12 @@ variable "subnet_id" {
   default     = "" # TODO remove as soon as subnet_id_runners and subnet_ids_gitlab_runner are gone. Variable is mandatory now.
 }
 
+variable "subnet_ids" {
```

Comment:
Just a quick response, need to dig futher in the details.

This variable seems similar to `subnet_id`. The variable you introduced here was used in the pas. But since the agent is is only deployed once it was refactored to single subnet. 

The subnets here are meant for the runner executors and not for the agent as far I can see. Maybe we should think about a better name to link better to the runner executors. 

In the comment you mention the firs tis used. I think this is not right. In another module I maintain I use the spot fleet API as well. Here we provide the api a list of subnets, instance types and price strategy. The fleet api creates instances matching this requirements.

Reply:
I think my comment is not explicit enough, all the subnets in the variable are used for agents, but only the first one is used for the runner itself since it can only be in one subnet. Maybe the best strategy here is to used one variable for the subnet of the runner.